numpy/lib/npyio.py

```diff
@@ -624,29 +634,15 @@ def _savez(file, args, kwds, compress, allow_pickle=True, pickle_kwargs=None):
     else:
         compression = zipfile.ZIP_STORED
 
-    zipf = zipfile_factory(file, mode="w", compression=compression)
-
-    # Stage arrays in a temporary file on disk, before writing to zip.
-    fd, tmpfile = tempfile.mkstemp(suffix='-numpy.npy')
-    os.close(fd)
-    try:
+    with zipfile_factory(file, mode="w", compression=compression) as zipf:
```

Comment:
As far as I known, as the exception handler was only here to remove/close files in the original version, no need to add try/except mechanism to the with statement.